src/AWS.Deploy.CLI/Commands/DeployCommand.cs

```diff
@@ -272,7 +272,12 @@ private async Task<Recommendation> GetSelectedRecommendationFromPreviousDeployme
 
             if (selectedRecommendation == null)
             {
-                var errorMessage = $"{deployedApplication.StackName} already exists as a Cloudformation stack but the recommendation used to deploy to the stack was not found.";
+                //check for persisted custom deployment recommendations
+                selectedRecommendation = recommendations.FirstOrDefault(x => string.Equals(x.Recipe.BaseRecipeId, deployedApplication.RecipeId, StringComparison.InvariantCultureIgnoreCase));

```

Comment:
If you change your `IsCompatible` function to be `GetRecommendation` and have it return back the recommendation instead of a bool you can use it here as well and replace likes 271-277 to be just

```
var selectedRecommenation = GetRecommendation(deployedApplication.RecipeId, recommendations);
```

In the `DeployedApplicationQueryer` that is currently calling `IsCompatible` you can change that to to call `GetRecommendation` and check to see if it returns back null.

Reply:
Fixed